r53dyndns.py

```diff
@@ -165,16 +170,27 @@ def main():
         current_ip = get_current_public_ip()
 
     hosted_zone_id = get_zone(zone_to_update)
-    record_ip = get_record_ip(hosted_zone_id, record_to_update)
 
-    if current_ip == record_ip:
-        print('IP addresses match - nothing to do')
+    record_ip = get_record_value(hosted_zone_id, record_to_update, 'A')
+    record_cname_value = get_record_value(hosted_zone_id, record_to_update, 'CNAME')
+
+    if current_ip == record_ip and record_cname_value == record_to_update:
+        print('Records match - nothing to do')
         exit(0)
 
-    status = update_record(hosted_zone_id, record_to_update, current_ip)
+    status = update_record(hosted_zone_id, record_to_update, 'A', current_ip)
+
+    if status == 'INSYNC':
+        print(
+            f'Updated A record {record_to_update} in hosted zone {zone_to_update} ({hosted_zone_id}) from '
+            f'{record_ip} to {current_ip}')
+
+    status = update_record(hosted_zone_id, route53_cname_record, 'CNAME', record_to_update)
```

Comment:
wrap this in `if route_cname_record:` since the environment variable is optional